File: src/test/java/engineer/thomas_werner/euler/math/TriangularNumberSequenceTest.java
package engineer.thomas_werner.euler.math;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class TriangularNumberSequenceTest {

    @Test
    public void testSequence() {
        final TriangularNumberSequence sequence = new TriangularNumberSequence();
        Assertions.assertEquals(1, sequence.nextValue().intValue());
        Assertions.assertEquals(3, sequence.nextValue().intValue());
        Assertions.assertEquals(6, sequence.nextValue().intValue());
        Assertions.assertEquals(10, sequence.nextValue().intValue());
        Assertions.assertEquals(15, sequence.nextValue().intValue());
        Assertions.assertEquals(21, sequence.nextValue().intValue());
    }

}

File: src/test/kotlin/de/huddeldaddel/euler/math/TriangularNumberSequenceTest.kt
package de.huddeldaddel.euler.math

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class TriangularNumberSequenceTest {

    @Test
    fun testSequence() {
        val sequence = TriangularNumberSequence()
        Assertions.assertEquals(1, sequence.nextValue().toInt())
        Assertions.assertEquals(3, sequence.nextValue().toInt())
        Assertions.assertEquals(6, sequence.nextValue().toInt())
        Assertions.assertEquals(10, sequence.nextValue().toInt())
        Assertions.assertEquals(15, sequence.nextValue().toInt())
        Assertions.assertEquals(21, sequence.nextValue().toInt())
    }

}
